Tests/SwiftNavigationTests/UITransactionTests.swift

```diff
@@ -2,81 +2,95 @@ import SwiftNavigation
 import XCTest
 
 class UITransactionTests: XCTestCase {
-  @MainActor
-  func testTransactionKeyPropagates() async {
-    let expectation = expectation(description: "onChange")
-    expectation.expectedFulfillmentCount = 2
-
-    let model = Model()
-    XCTAssertEqual(UITransaction.current.isSet, false)
-
-    observe {
-      if model.count == 0 {
-        XCTAssertEqual(UITransaction.current.isSet, false)
-      } else if model.count == 1 {
-        XCTAssertEqual(UITransaction.current.isSet, true)
-      } else {
-        XCTFail()
+  #if compiler(>=6)
```

Comment:
I was a little torn by this. Technically this test is not running on Apple platforms in CI since we aren't dealing with any Xcode betas right now. But we are getting coverage as far as Wasm is concerned since we are running a Swift 6 snapshot. And afaict there isn't a way to write a test that can run on both Xcode 15.4 + Wasm.

We _could_ copy-paste this test to make a version that works on Xcode 15.4. I went with this approach since hopefully soon we will be able to target Xcode 16 in CI, at which point this is all moot.